### ISSUE TITLE
Ignoring properties of nested lists

### ISSUE THREAD

**Issue by joshhubers** [ValeraT1982/ObjectsComparer#7]
Great library!

I was wondering if you could help me out. I'm trying to ignore a property on a nested list (specifically the List<T>.Capacity property), and can't quite seem to solve it.

This is a quick example:

```
public class Program
  {
    public static void Main(string[] args)
    {
      List<Foo> foosOne = new List<Foo>
      {
        new Foo { Bars = new List<Bar> { new Bar { A = 1, B = "foo" }, new Bar { A = 2, B = "bar" }}},
      };

      List<Foo> foosTwo = new List<Foo>
      {
        new Foo { Bars = new List<Bar> { new Bar { A = 1, B = "foo" }, new Bar { A = 2, B = "bar" }}},
      };

      foosOne[0].Bars.TrimExcess();

      var comparer = new ObjectsComparer.Comparer<IEnumerable<Foo>>();

      comparer.AddComparerOverride(
        // Not right, but I'm unsure what to do here to make this work
        () => new Foo().Bars.Capacity, 
        (listOne, listTwo, settings) => true );

      IEnumerable<Difference> differences;
      var isEqual = comparer.Compare(foosOne, foosTwo, out differences);
        
      foreach(var diff in differences)
      {
        Console.WriteLine(diff);
      }
    }
  }

  public class Foo
  {
    public List<Bar> Bars { get; set; }
  }

  public class Bar
  {
    public int A { get; set; }
    public string B { get; set; }
  }
```

This produces a difference:
```
Difference: DifferenceType=ValueMismatch, MemberPath='[0].bars.Capacity', Value1='2', Value2='4'.
```

If it matters I would also like to make it generic so that I can ignore capacity of any models with nested lists.

**Comment by ValeraT1982** [comment]
Very interesting case. It's an issue. It will be fixed in the next version and will work without adding override for this property.

Short term solution:

```
comparer.AddComparerOverride(
        () => new List<Bar>.Capacity, 
        DoNotCompareValueComparer.Instance);
```

**Comment by joshhubers** [comment]
Alright.

For now I'll have to do a separate workaround as I have many models with different nested list types, and it's not practical to define an override for every nested list property.

Ex. "in reality" I have many different ```Foo``` types and each ```Foo``` is more like:
```
  public class Foo
  {
    public List<Bar> Bars { get; set; }
    public List<Fizz> Fizzez { get; set; }
    public List<Buzz> Buzzez { get; set; }
  }
```

Thanks for your work on this!

**Comment by ValeraT1982** [comment]
It's fixed in this repository now. You can download, build and use it. 
I'm going to upload new version in 1-2 months.

**Comment by joshhubers** [comment]
Fantastic!

Thanks!

**Comment by ValeraT1982** [comment]
Fixed in 1.3
